tools/generate_readmes.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
```

Comment:
Yoseph,
this is quite a long PR. It would be good to break the changes and send them incrementally. 
I'll abide to the preference of the group, but that's my suggestion.

Also, we should discuss the format of documentation for the samples.